### PR TITLE
Feature/phone service insert event

### PR DESCRIPTION
**Pull request by ShenDing1125**
Download data from Redis and store it in the DB.

### CODE REVIEW

File: src/runner/porter.runner.ts [hunk 1]
```diff
@@ -173,6 +175,116 @@ export class PorterRunner extends Runner {
...
+        if (!hasDataProperty(data)) {
+          errMesInit.info = 'Data format error';
+          errAry.push(errMesInit);
+          continue;
+        }
+        const { memberNotes, lastMemberNotes } = data;
+        const { criteria, lastMemberRecord } = lastMemberNotes;
+
+        try {
+          await manager.getRepository(MemberNote).insert(memberNotes);
```
Comment (TedxTed) [inline src/runner/porter.runner.ts]: we usually handle all data processing and logic within `.infra.ts` files
this approach helps us keep our data handling consistent and maintainable across the repo

File: src/runner/porter.runner.ts [hunk 2]
```diff
@@ -173,6 +175,116 @@ export class PorterRunner extends Runner {
...
+    const pattern = `PhoneService:*`;
+    const errAry: ErrMesType[] = [];
+    let cursor = '0';
+
+    do {
+      const scanResult = await client.scan(cursor, 'MATCH', pattern, 'COUNT', batchSize);
+      cursor = scanResult[0];
+      const keys = scanResult[1];
+
+      for (const key of keys) {
```
Comment (TedxTed) [inline src/runner/porter.runner.ts:226]: In this `for (const key of keys)` loop, let's be cautious of potential N+1 problems and consider bulk operations  to manage performance and runner time limits.

File: src/runner/porter.runner.ts [hunk 3]
```diff
@@ -173,6 +175,116 @@ export class PorterRunner extends Runner {
...
+        }
+
+        await client.del(key);
+      }
+    } while (cursor !== '0');
+
+    if (errAry.length > 0) {
+      for (const item of errAry) {
+        console.error(`Saving phone service failed:${item}`);
+        console.log(item);
```
Comment (TedxTed) [inline src/runner/porter.runner.ts]: looks like we have an extra console.log(item) here

File: src/runner/porter.runner.ts [hunk 4]
```diff
@@ -173,6 +175,116 @@ export class PorterRunner extends Runner {
...
+          continue;
+        }
+        const { memberNotes, lastMemberNotes } = data;
+        const { criteria, lastMemberRecord } = lastMemberNotes;
+
+        try {
+          await manager.getRepository(MemberNote).insert(memberNotes);
+        } catch (error) {
+          if (typeof errMesInit.info !== 'string' && errMesInit.info.memberNote === 'NoError') {
+            errMesInit.info.memberNote = { data: memberNotes, errMsg: error.toString() };
```
Comment (TedxTed) [inline src/runner/porter.runner.ts]: It seems the current catch block's condition` if (typeof errMesInit.info !== 'string' && errMesInit.info.member === 'NoError') ` might miss catching some unexpected errors

**Review by TedxTed — CHANGES_REQUESTED** [review]

File: src/runner/porter.runner.ts (hunk 1 above)
Comment (ShenDing1125) [reply src/runner/porter.runner.ts]: Already changed the location. Thank you for the reminder.

File: src/runner/porter.runner.ts (hunk 3 above)
Comment (ShenDing1125) [reply src/runner/porter.runner.ts]: Already removed, thanks.

File: src/runner/porter.runner.ts (hunk 4 above)
Comment (ShenDing1125) [reply src/runner/porter.runner.ts]: I have changed the code to a new implementation, avoiding misunderstanding that could lead to skipping error handling due to the 'if' statement.

**Comment by clothe09986** [comment]
[(5) 【聯絡紀錄】搬去lodestar-app-backend的 PorterRunner處理 [4]](https://trello.com/c/kjUVX1T6/4248-5-%E3%80%90%E8%81%AF%E7%B5%A1%E7%B4%80%E9%8C%84%E3%80%91%E6%90%AC%E5%8E%BBlodestar-app-backend%E7%9A%84-porterrunner%E8%99%95%E7%90%86-4)

**Review by TedxTed — APPROVED** [review]

**Review by Ponchimeow — APPROVED** [review]

File: src/runner/porter.runner.ts (hunk 3 above)
Comment (kkshyu) [reply src/runner/porter.runner.ts]: NITS: we can add this config into our linting file
https://stackoverflow.com/a/49990579

**Pull request closed without merging**